Core/include/Acts/Fitter/KalmanFitter.hpp

```diff
@@ -70,13 +72,16 @@ struct KalmanFitterOptions {
                       std::reference_wrapper<const MagneticFieldContext> mctx,
                       std::reference_wrapper<const CalibrationContext> cctx,
                       const OutlierFinder& outlierFinder_,
-                      LoggerWrapper logger_, const Surface* rSurface = nullptr,
+                      LoggerWrapper logger_,
+                      const PropagatorPlainOptions& pOptions,
```

Comment:
This allows the configuration of regular options but afterwards it still wont be possible to use the dense stepping in the (C)KF. Your approach solves the problem but if I think about the dense issue then there needs to be a follow-up PR for it. Assuming that this would then be solved by passing the propagator options directly as an argument here, this could make your current changes unnecessary. So my question is: Could this argument be replaced by some `propagator_options_t` or is there another solution such that these changes remain necessary?

Reply:
I have an idea. It looks like the `DenseStepperPropagatorOptions` have three additional options which might be set by the user, i.e. the `meanEnergyLoss`, `includeGgradient` and `momentumCutOff`. So if we include those three additional options in the `PropagatorPlainOptions` and reimplement the `setPlainOptions` method in the `DenseStepperPropagatorOptions`, we could configure the options for the dense stepper as well. What do you think?

Reply:
I thought about this approach back then but this might lead to a struct that will have tons of parameters in the future and you are not sure anymore which of them are actually used or not. That's why I decided to split it into these two classes. Then one has to really use the right options in order to compile it. The current solution is not a final one but solved the problem back then. Any suggestions are welcome.

Reply:
Yeah. If we have more propagator options, it could be a mess. In that case, I agree that we should explicitly include the propagator options in the (C)KF, which might make the (C)KF `fit` and `findTracks` interface a bit more complicated to set up from the user point of view, but should remove all the mess.